### ISSUE TITLE
Android UI not responding 

### ISSUE THREAD

**Issue by mbalajee** [google/filament#7956]
⚠️ **Issues not using this template will be systematically closed.**

**Describe the bug**
When loading larger models (over 8K entities) `ModelViewer#loadModelGlb` blocks the UI.  Calling this function in a different thread results in  `This thread has not been adopted` when executing `ResourceLoader::Impl::computeTangents`

**To Reproduce**
Loading a larger model hangs the entire UI, can't exit out of the screen

**Expected behavior**
UI should not be blocked

**Desktop (please complete the following information):**
 - OS: Android
```
FEngine resolved backend: OpenGL
QUALCOMM build                   : 8e5405b, I57aaec3440
Build Date                       : 05/21/21
OpenGL ES Shader Compiler Version: EV031.32.02.10
Local Branch                     : mybranchebba1dbe-451b-f160-ac81-1458d0b52ae8
Remote Branch                    : quic/gfx-adreno.lnx.1.0.r135-rel
Remote Branch                    : NONE
Reconstruct Branch               : NOTHING
Build Config                     : S P 10.0.7 AArch64
Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
PFP: 0x016ee190, ME: 0x00000000
[Qualcomm], [Adreno (TM) 630], [OpenGL ES 3.2 V@0502.0 (GIT@8e5405b, I57aaec3440, 1621601959) (Date:05/21/21)], [OpenGL ES GLSL ES 3.20]
```

**Smartphone (please complete the following information):**
 - Device: OnePlus 6T/6GB RAM
 - OS: Android 11

**Additional context**
Sorry couldn't attach the glb file as it is confidential

**Comment by mbalajee** [comment]
UI is not blocked when I create the `ModelViewer` instance (& so the `Engine` instance) in `Executors.newSingleThreadExecutor()` & call `modelViewer.render` from the same executor.  Is this the right way to unblock UI or is there a downside in creating ModelViewer in a non-main thread?

**Comment by mbalajee** [comment]
Oh when I back out of the screen I get this error,
```
Precondition
in shutdown:444
reason: Engine::shutdown() called from the wrong thread!
```
because ModelViewer destroys the view in `addOnAttachStateChangeListener` so I guess it is not as straightforward as operating modelViewer on a non-ui thread

@romainguy would you recommend maintaining a copy of ModelViewer ?

**Comment by innolight** [comment]
@mbalajee we faced the same issue as you do. Are you aware of any workarounds for the issue?

@romainguy @pixelflinger this issue is blocking us from adopting this library in production. Do you have any suggestion for this performance issue?

**Comment by mbalajee** [comment]
@innolight Yeah we ended up using Filament in a different thread from Android side.  All calls to filament must be executed on the same thread.  So UI is not blocked while Filament does its job initially.
